File: javascript-closure-angular/API/Client/ResponseRoutesResult.js
goog.provide('API.Client.ResponseRoutesResult');

/**
 * @record
 */
API.Client.ResponseRoutesResult = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.ResponseRoutesResult.prototype.searchId;

/**
 * @type {!Array<!API.Client.ResponseRoutesLocation>}
 * @export
 */
API.Client.ResponseRoutesResult.prototype.locations;

/**
 * @type {!Array<!string>}
 * @export
 */
API.Client.ResponseRoutesResult.prototype.unreachable;

File: scalaz/src/main/scala/org/openapitools/client/api/ResponseRoutesResult.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ResponseRoutesResult._

case class ResponseRoutesResult (
  searchId: String,
locations: List[ResponseRoutesLocation],
unreachable: List[String])

object ResponseRoutesResult {
  import DateTimeCodecs._

  implicit val ResponseRoutesResultCodecJson: CodecJson[ResponseRoutesResult] = CodecJson.derive[ResponseRoutesResult]
  implicit val ResponseRoutesResultDecoder: EntityDecoder[ResponseRoutesResult] = jsonOf[ResponseRoutesResult]
  implicit val ResponseRoutesResultEncoder: EntityEncoder[ResponseRoutesResult] = jsonEncoderOf[ResponseRoutesResult]
}
